src/graph.cpp

```diff
@@ -192,21 +187,22 @@ double Graph::findBottleneck(const std::string &target) const {
  * @param target - Id of the target Vertex
  * @param value - Number of units to augment the flow by
  */
-void Graph::augmentPath(const std::string &target, const unsigned int &value) const {
+void Graph::augmentPath(const std::string &target, const unsigned int &value, Graph &regularGraph) const {
     Vertex *currentVertex = findVertex(target);
 
     while (currentVertex->getPath() != nullptr) {
-        Edge *currentPath = currentVertex->getPath();
+        Edge *residualEdge = currentVertex->getPath();
+        Edge *reverseResidualEdge = residualEdge->getReverse();
+        Edge *regularEdge = getCorrespondingEdge(residualEdge, regularGraph);
 
-        if (currentPath->getDest() == currentVertex) //Regular Edge
-        {
-            currentPath->setFlowValue(*(currentPath->getFlow()) + value);
-            currentVertex = currentVertex->getPath()->getOrig();
+        //Augment flow in regular graph
+        regularEdge->setFlow(regularEdge->getFlow() + value);
 
-        } else { //Residual Edge (was traversed backwards)
-            currentPath->setFlowValue(*(currentPath->getFlow()) - value);
-            currentVertex = currentVertex->getPath()->getDest();
-        }
+        //Update residual Graph edges
+        residualEdge->setCapacity(regularEdge->getCapacity() - regularEdge->getFlow());
+        reverseResidualEdge->setCapacity(reverseResidualEdge->getCapacity() + value);
```

Comment:
Shouldn't we be working in terms of flux instead of capacity and let the capacity be a constant value? It ends up the same, since the difference between flux and capacity is what matters. But it's easier to keep up with that way

Reply:
It's a matter of convention, because in the residual graph we only use one value, it is most often referred to as residual capacity.